### PR TITLE
DOCS: Move undefined FAQ item

### PR DESCRIPTION
**Pull request by ChrisChinchilla**
### Description

Move undefined FAQ item into main docs, breaking https://github.com/ethereum/solidity/pull/4499/files

### Checklist
- [x] Code compiles correctly
- [x] All tests are passing
- [x] New tests have been created which fail without the change (if possible)
- [x] README / documentation was extended, if necessary
- [x] Changelog entry (if change is visible to the user)
- [x] Used meaningful commit messages

### CODE REVIEW

File: docs/types.rst [hunk 1]
```diff
@@ -13,6 +13,8 @@ Solidity provides several elementary types which can be combined to form complex
 In addition, types can interact with each other in expressions containing
 operators. For a quick reference of the various operators, see :ref:`order`.
 
+The concept of "undefined" or "null" values do not exist in Solidity. To handle any unexpected values, you should use the :ref:`revert function<assert-and-require>` to revert the whole transaction.
```
Comment (chriseth) [inline docs/types.rst]:
```suggestion
The concept of "undefined" or "null" values do not exist in Solidity. To handle any unexpected values, you should use the :ref:`revert function<assert-and-require>` to revert the whole transaction, or return a tuple with a second `bool` value denoting success.
```

File: docs/types.rst [hunk 2]
```diff
@@ -13,6 +13,8 @@ Solidity provides several elementary types which can be combined to form complex
...
+
```
Comment (chriseth) [inline docs/types.rst:20]: Something we should perhaps also mention here: Freshly declared Solidity variables always have a default value - there is no undefined behaviour. The default value depends on the type, but is usually straightforward (zero, false, empty, etc).

**Review by chriseth — CHANGES_REQUESTED** [review]

File: docs/types.rst (hunk 2 above)
Comment (ChrisChinchilla) [reply docs/types.rst:20]: @chriseth Added details to where reader can find more details on default values.

File: docs/types.rst [hunk 3]
```diff
@@ -13,6 +13,11 @@ Solidity provides several elementary types which can be combined to form complex
 In addition, types can interact with each other in expressions containing
 operators. For a quick reference of the various operators, see :ref:`order`.
 
+The concept of "undefined" or "null" values do not exist in Solidity, but newly
```
Comment (erak) [inline docs/types.rst]:
```suggestion
The concept of "undefined" or "null" values does not exist in Solidity, but newly
```

**Review by erak — APPROVED** [review]

**Review by chriseth — APPROVED** [review]

**Pull request merged**